include/pybind11/detail/internals.h

```diff
@@ -154,7 +157,7 @@ struct type_info {
 };
 
 /// Tracks the `internals` and `type_info` ABI version independent of the main library version
-#define PYBIND11_INTERNALS_VERSION 4
+#define PYBIND11_INTERNALS_VERSION 5
```

Comment:
If we are bumping the ABI, we should probably move through several of the PRs sitting round that bump the ABI and get them in too. Also, this is going to create a massive mess, as we'll have to start a conda-forge migration, and quite a few packages don't use the conda-forge's pybind11, so...  It will be a mess. But I guess it's going to have to happen.

Reply:
There may be a non-abi bumping strategy available; if you like I can take that approach.